odfdom/src/main/java/org/odftoolkit/odfdom/doc/table/OdfTableCell.java

```diff
@@ -507,13 +507,13 @@ public TableTableCellElementBase getOdfElement() {
    *     is not "currency".
    */
   public String getCurrencyCode() {
-    if (mCellElement
-        .getOfficeValueTypeAttribute()
-        .equals(OfficeValueTypeAttribute.Value.CURRENCY.toString())) {
-      return mCellElement.getOfficeCurrencyAttribute();
-    } else {
-      throw new IllegalArgumentException();
+    String t = mCellElement.getOfficeValueTypeAttribute();
```

Comment:
How about simply doing this slightly shorter and a message that also shows the unexpected value?

```
if (!Objects.equals(t, OfficeValueTypeAttribute.Value.CURRENCY.toString()) {
    LOG.log(Level.WARNING,"@office:value-type attribute expected to be of type CURRENCY but was {0}", t);
}

```

Reply:
Cool! I was not aware of the null-safe _Objects.equals(..)_ will surely overtake it, and I do like your improvement of giving out what the incorrect value is! Thanks!😎👍